### PR TITLE
Intermediate.ATX.2017Q1.James.Puckett

### PR DESCRIPTION
**Pull request by JamesPuckett**
# Checkpoint Rubric

This is the rubric that your instructor will use to grade your checkpoints. Please do not edit. 

## Checkpoint 1: Towers of Hanoi Terminal App
- [X] All tests passed: 55 points
- [ ] Properly commented code: 15 points
0
- [X] Properly indented code: 15 points
- [ ] Demonstrated effective use of JavaScript: 15 points
10/15

Total: 80

Specific Comments:
See the github file comments for details

*Comments may also be made on the file itself using github's commenting tools.*

## Checkpoint 2: Towers of Hanoi GUI App
- [X] All tests passed: 55 points
- [ ] Properly commented code: 15 points
small ding because of checked-in commented code: 13/15
- [X] Properly indented code: 15 points
- [X] Demonstrated effective use of jQuery: 15 points

98/100

Specific Comments:

*Comments may also be made on the file itself using github's commenting tools.*

## Checkpoint 3: Final Project Gist Blog
- [X] Effective use of AJAX: 20 points
- [X] Effective use of loops: 20 points
- [X] Effective use of github API: 20 points
15/20 - looks like a slight issue with the gist filename
- [X] Style and design: 10 points
- [X] Properly commented code: 15 points
good comments
- [X] Properly indented code: 15 points

Specific Comments:
Well done! 95/100

*Comments may also be made on the file itself using github's commenting tools.*

### CODE REVIEW

File: apps/01PigLatin.js [hunk 1]
```diff
@@ -7,17 +7,109 @@ var rl = readline.createInterface({
...
+        console.log('restWord: ' + restWord);
+        //returns the final word
+        return restWord + firstPart + 'ay';
+      }
+
+    }
+  }
+
+  //decided all of this was unnessary and commented it out.*************
+
```
Comment (zacacollier) [inline apps/01PigLatin.js]: Good call - D.R.Y :)

File: apps/01PigLatin.js [hunk 2]
```diff
@@ -7,17 +7,109 @@ var rl = readline.createInterface({
...
+  var vowel = '';
+  //checks to see if it starts with a vowel and if it does it adds 'yay' to the end
+  if (word.slice(0, 1) === 'a' || word.slice(0, 1) === 'e' || word.slice(0, 1) === 'i' || word.slice(0, 1) === 'o' || word.slice(0, 1) === 'u' || word.slice(0, 1) === 'y') {
+    return word + 'yay';
+  }
+  //loops through 'word' and finds the first vowel
+  for (var i = 0; i < word.length; i++) {
+    if (word[i] === 'a' || word[i] === 'e' || word[i] === 'i' || word[i] === 'o' || word[i] === 'u' || word[i] === 'y') {
+      //stores first vowel here
+      vowel = word.slice(i, i + 1);
```
Comment (zacacollier) [inline apps/01PigLatin.js:30]: nicely done!

File: apps/02RockPaperScissors.js [hunk 3]
```diff
@@ -8,16 +8,109 @@ var rl = readline.createInterface({
...
+      getPrompt();
+    }
+    if (play2 === 2) {
+      console.log('Hand 2 wins 2 out of 3! Reseting...');
+      p1Wins = 0;
+      p2Wins = 0;
+      tie = 0;
+      getPrompt();
+    }
+  }
```
Comment (zacacollier) [inline apps/02RockPaperScissors.js:95]: I like these features you added, very cool!

File: apps/03TicTacToe.js [hunk 4]
```diff
@@ -24,23 +24,59 @@ function printBoard() {
...
 }
 
 function diagonalWin() {
-  // Your code here
+  if ((board[0][0] === playerTurn && board[1][1] === playerTurn && board[2][2] === playerTurn) || (board[0][2] === playerTurn && board[1][1] === playerTurn && board[2][0] === playerTurn)) {
+    return true;
+  } else {
+    return false;
+  }
 }
```
Comment (RX7RUSS) [inline apps/03TicTacToe.js:52]: I like this solution to reset the board. Integrating it into the checkForWin function and logging a message to the user that the board is being reset is nice and clean.

File: apps/01PigLatin.js [hunk 5]
```diff
@@ -8,31 +8,43 @@ var rl = readline.createInterface({
...
+  //stores first part of word before vowel
   var firstPart = '';
+  //stores vowel and rest of word
   var restWord = '';
+  //stores only the vowel
   var vowel = '';
+  //checks to see if it starts with a vowel and if it does it adds 'yay' to the end
   if (word.slice(0, 1) === 'a' || word.slice(0, 1) === 'e' || word.slice(0, 1) === 'i' || word.slice(0, 1) === 'o' || word.slice(0, 1) === 'u' || word.slice(0, 1) === 'y') {
     return word + 'yay';
   }
```
Comment (RX7RUSS) [inline apps/01PigLatin.js]: This is a great solution for finding the vowels. Much better than having individual if statements for each vowel.

File: apps/03TicTacToe.js [hunk 6]
```diff
@@ -24,23 +27,89 @@ function printBoard() {
...
 
 function horizontalWin() {
-  // Your code here
+  if ((board[0][0] === playerTurn && board[0][1] === playerTurn && board[0][2] === playerTurn) || (board[1][0] === playerTurn && board[1][1] === playerTurn && board[1][2] === playerTurn) || (board[2][0] === playerTurn && board[2][1] === playerTurn && board[2][2] === playerTurn)) {
+    return true;
+  } else {
+    return false;
```
Comment (tjefferson08) [inline apps/03TicTacToe.js:33]: Yay for keeping data types consistent!

File: apps/04TowersOfHanoi.js [hunk 7]
```diff
@@ -19,23 +19,59 @@ function printStacks() {
...
 
-function movePiece() {
-  // Your code here
-
+function movePiece(startStack, endStack) {
+  var starting = stacks[startStack];
+  var startPiece = starting.pop();
+  var end = stacks[endStack];
+  var ending = null;
+  ending = end.push(startPiece);
```
Comment (tjefferson08) [inline apps/04TowersOfHanoi.js:27]: What is an example of a value that `ending` might get here?

File: apps/04TowersOfHanoi.js [hunk 8]
```diff
@@ -19,23 +19,59 @@ function printStacks() {
...
 }
 
-function isLegal() {
-  // Your code here
+function isLegal(startStack, endStack) {
+  var starting = stacks[startStack];
+  var end = stacks[endStack];
+  var ending = null;
```
Comment (tjefferson08) [inline apps/04TowersOfHanoi.js:33]: Does this guy ever get used?

File: apps/04TowersOfHanoi.js [hunk 9]
```diff
@@ -19,23 +19,59 @@ function printStacks() {
...
+  if (starting[starting.length - 1] > end[end.length - 1]) {
```
Comment (tjefferson08) [inline apps/04TowersOfHanoi.js:34]: What if start/end arrays are empty? What values will this comparison be comparing?

File: apps/04TowersOfHanoi.js [hunk 10]
```diff
@@ -19,23 +19,59 @@ function printStacks() {
...
+    console.log("invalid move.");
+    return false;
+  } else {
+    return true;
+  }
+}
 
+function verifyInput(startStack, endStack) {
```
Comment (tjefferson08) [inline apps/04TowersOfHanoi.js:42]: Nice initiative!

File: apps/04TowersOfHanoi.js [hunk 11]
```diff
@@ -19,23 +19,59 @@ function printStacks() {
...
+    return false;
+  }
 }
 
 function checkForWin() {
-  // Your code here
-
+  if (stacks.b.length === 4 || stacks.c.length === 4) {
+    console.log('you win');
+    stacks = {
```
Comment (tjefferson08) [inline apps/04TowersOfHanoi.js:54]: Cool idea here!

File: apps/04TowersOfHanoi.js [hunk 12]
```diff
@@ -19,23 +19,59 @@ function printStacks() {
...
+    return false;
+  }
 }
 
 function towersOfHanoi(startStack, endStack) {
-  // Your code here
+  startStack = startStack.toLowerCase();
+  endStack = endStack.toLowerCase();
+  if (verifyInput(startStack, endStack)) {
+    if (isLegal(startStack, endStack)) {
```
Comment (tjefferson08) [inline apps/04TowersOfHanoi.js:69]: Nice extra step here! I think we can clean this up even more by combining the `if`s with a `&&`

File: apps/08jquery-towers-of-hanoi/script.js [hunk 13]
```diff
@@ -1,5 +1,37 @@
 'use strict';
-
-$(document).ready(function() {
+var $block = null;
+$(document).ready(function () {
   // Put app logic here
+    $('[data-stack]').click(function () {
+      //picks up the block
+      if (!$block){
```
Comment (tjefferson08) [inline apps/08jquery-towers-of-hanoi/script.js:7]: This if/else reads like this:
"IF NOT block, then pick it up, ELSE put it down"
The double negative in the else case is a little confusing, IMO

I'd prefer to see:
"IF block, put it down, ELSE pick it up"

Just a smidge easier to read

File: apps/08jquery-towers-of-hanoi/script.js [hunk 14]
```diff
@@ -1,5 +1,37 @@
...
+        $block = $(this).children().last().detach();
+        $('div#announce-game-won').empty();
+      } 
+      //puts down the block if holding the block
+      else {
+        if($(this).children().last().data('block')>$block.data('block') || $(this).children().length === 0){
```
Comment (tjefferson08) [inline apps/08jquery-towers-of-hanoi/script.js:13]: nitpick: add spaces around the `>`

File: apps/08jquery-towers-of-hanoi/script.js [hunk 15]
```diff
@@ -1,5 +1,37 @@
...
+          $(this).append($block);
+          $block = null;
+          checkForWin();
+        }else {
+          $('div#announce-game-won').text('That spot is invalid.');
```
Comment (tjefferson08) [inline apps/08jquery-towers-of-hanoi/script.js:18]: Nice feature!

File: apps/08jquery-towers-of-hanoi/script.js [hunk 16]
```diff
@@ -1,5 +1,37 @@
...
+      }else {
+        return false;
+      }
+    }
 });
+
+
+
+
+//$('[data-stack="2"]').append($block)
```
Comment (tjefferson08) [inline apps/08jquery-towers-of-hanoi/script.js:35]: It's good practice to remove commented code before pushing

File: apps/11gist-blog/script.js [hunk 17]
```diff
@@ -1,5 +1,58 @@
 'use strict';
 
-$(document).ready(function() {
-  // You code here
+$(document).ready(function () {
+  $.ajax('https://api.github.com/users/JamesPuckett/gists', { //http://127.0.0.1:8080/apps/11gist-blog/api/gists.json
```
Comment (tjefferson08) [inline apps/11gist-blog/script.js:4]: woo real github!

File: apps/11gist-blog/script.js [hunk 18]
```diff
@@ -1,5 +1,58 @@
...
+    success: function (response) {
+      response.forEach(function (post) {
+        var $description = post.description;
+        var $url = post.url;
+        var $comments = post.comments_url;
+        //only uses posts that start with '#post ' and removes the string '#post '
+        //creates a link that has to data attritubutes so i can reference them later with ajax calls
+        if ($description.substring(0, 6) === '#post ') {
+          var $str = `
```
Comment (tjefferson08) [inline apps/11gist-blog/script.js:13]: Yessss template strings. nice

File: apps/11gist-blog/script.js [hunk 19]
```diff
@@ -1,5 +1,58 @@
...
+          <li>
+            <a href='#' data-comments=${$comments} data-url=${$url}>${$description.replace('#post ', '')}
+            </a>
+          </li>
+          `;
+          $('#posts').append($str);
+        }
+        //this section below here does not appear to be working.
```
Comment (tjefferson08) [inline apps/11gist-blog/script.js:21]: Looks like the issue you're having is that the title of the file is expected to be 'post.md' but yours is coming back as gistfile1.txt

File: apps/11gist-blog/script.js [hunk 20]
```diff
@@ -1,5 +1,58 @@
...
+      })
+      $('#posts a').on('click', function (link) {
```
Comment (tjefferson08) [inline apps/11gist-blog/script.js:23]: good choice of selector

File: apps/11gist-blog/script.js [hunk 21]
```diff
@@ -1,5 +1,58 @@
...
+            $('#post').empty();
+            $('#comments').empty();
+            var $content = marked(clicked['files']['post.md']['content']);
+            $('#post').append($content);
+          }
+        })
+        $.ajax($(this).data('comments'), {
+          success: function (commentFetch) {
+            commentFetch.forEach(function (comment) {
+              var $userLogin = comment.user.login;
```
Comment (tjefferson08) [inline apps/11gist-blog/script.js:38]: good names here

**Pull request closed without merging**